flax/linen/module.py

```diff
@@ -341,17 +348,9 @@ def __post_init__(self):
     else:
       raise ValueError("parent must be None, Module or Scope")
 
-    # Call the user-defined initialization function.
-    self.setup()
-
-  def setup(self):
-    """Called when Module instance receives variables and PRNGs.
-
-    If you want to use PRNGs and/or read or write variables during module
-    instance construction, override this method. It will be automatically
-    called from the dataclass `__post_init__`.
-    """
-    pass
+    # Call the user-defined initialization setup() function.
```

Comment:
I don't like this change. Where would we put the docstring for setup()?

Reply:
restored it.